frontend/src/features/routes/routesSlice.ts

```diff
@@ -0,0 +1,40 @@
+import apiSlice from "../../app/apiSlice";
+import { type Coordinate } from "../location/locationSlice";
+
+/**
+ * A list of routes, as defined by the backend.
+ */
+export type Routes = Route[];
+
+/**
+ * A Route, as defined by the backend.
+ */
+export interface Route {
+  id: string;
+  name: string;
+  stopIds: number[];
+  waypoints: Coordinate[];
+  isActive: boolean;
+}
+
+// --- API Definition ---
+
+/**
+ * This slice extends the existing API slice with the route information route.
+ * There should be no state to reason about here, it's just configuration.
+ */
+const routesApiSlice = apiSlice.injectEndpoints({
+  endpoints: (builder) => ({
+    // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+    getRoutes: builder.query<Routes, void>({
```

Comment:
for the eslint error we can use undefined or empty object {} instead of void I believe

Reply:
Not really, if we switched it to undefined/{} you would have to specify a useless value whenever you used the query hook. eslint doesn't like the void use but it allows us to avoid that.

Reply:
should we disable that rule then?